src/libsystemd/sd-journal/journal-file.c

```diff
@@ -3071,6 +3081,16 @@ static int generic_array_bisect(
 
                         for (;;) {
                                 if (left == right) {
+                                        if (m != m_original && direction == DIRECTION_DOWN) {
```

Comment:
Why only do this logic if we're going in the down direction? Doesn't this apply regardless of the direction we're going in?

Reply:
As 'left' is always tested when we are going to upwards. I extended the comment in the code.